gobcore/logging/logger.py

```diff
@@ -272,38 +287,15 @@ def configure(self, msg: dict, name: str = None):
                 'stepid': header.get('stepid')
             }
 
-    def add_message_broker_handler(self):
-        """Adds the message broker handler to the `name` instance if one doesn't exist yet."""
-        name = "RequestsHandler"
-
-        # make sure we have only 1 handler
-        if not any(
-            isinstance(handler, RequestsHandler) and handler.name == name
-            for handler in self.get_logger().handlers
-        ):
-            handler = RequestsHandler()
-            handler.name = name
-            formatter = logging.Formatter(self.LOGFORMAT)
-            handler.setFormatter(formatter)
-            self.get_logger().addHandler(handler)
-
-    def _init_logger(self):
-        if self.name in self._logger:
-            return
-
+    def _init_logger(self, handlers: list[type[logging.Handler]]):
         new_logger = logging.getLogger(self.name)
         new_logger.setLevel(self.LOGLEVEL)
 
-        # log default to stdout
-        new_logger.addHandler(logging.StreamHandler(stream=sys.stdout))
-        self.set_logger(new_logger)
-
-    def get_logger(self) -> logging.Logger:
-        return self._logger[self.name]
+        for handler in handlers:
+            handler = handler()
```

Comment:
I would leave the handler initialisation to the caller - in case the caller needs to do some extra initialisation. Makes this easier to read as well. This reads a bit weird.